File: src/python/PythonSDK/foundationallm/models/orchestration/knowledge_management_completion_request.py
"""
Class Name: KnowledgeManagementCompletionRequest
Description: Encapsulates the metadata required to complete a knowledge management orchestration request.
"""
from typing import List, Optional
from pydantic import BaseModel
from foundationallm.models.orchestration import MessageHistoryItem
from foundationallm.models.metadata import KnowledgeManagementAgent

class KnowledgeManagementCompletionRequest(BaseModel):
    """
    Orchestration completion request.
    """
    user_prompt: str
    agent: Optional[KnowledgeManagementAgent] = None   
    message_history: Optional[List[MessageHistoryItem]] = []

File: src/dotnet/Common/Models/Orchestration/KnowledgeManagementCompletionRequest.cs
using FoundationaLLM.Common.Models.Agents;
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Orchestration
{
    /// <summary>
    /// The Knowledge Management Completion Request model.
    /// </summary>
    public class KnowledgeManagementCompletionRequest : LLMCompletionRequest
    {
        /// <summary>
        /// The Knowledge Management agent metadata.
        /// </summary>
        [JsonPropertyName("agent")]
        public required KnowledgeManagementAgent Agent { get; set; }
    }
}
